src/vcpkg/postbuildlint.cpp

```diff
@@ -578,6 +581,26 @@ namespace vcpkg::PostBuildLint
             }
         }
 
+        if (!binaries_with_invalid_architecture.empty())
+        {
+            print_invalid_architecture_files(expected_architecture, binaries_with_invalid_architecture);
+            return LintStatus::ERROR_DETECTED;
+        }
+#elif defined(__APPLE__)
+        std::vector<FileAndArch> binaries_with_invalid_architecture;
```

Comment:
I think this would be better implemented as a separate function unless it can actually share some of the implementation.

Reply:
Yeah the function name was very matching and it is doing the, but now for mac. If you want I can change it, but I would prefer the current solution.